Mathlib/Order/Filter/Archimedean.lean

```diff
@@ -0,0 +1,254 @@
+
+theorem tendsto_nat_cast_atTop_atTop [StrictOrderedSemiring R] [Archimedean R] :
+    Tendsto ((↑) : ℕ → R) atTop atTop :=
+  tendsto_nat_cast_atTop_iff.2 tendsto_id
+#align tendsto_coe_nat_at_top_at_top tendsto_nat_cast_atTop_atTop
+
+@[simp] theorem Int.comap_cast_atTop [StrictOrderedRing R] [Archimedean R] :
+    comap ((↑) : ℤ → R) atTop = atTop :=
+  comap_embedding_atTop (fun _ _ => Int.cast_le) fun r =>
+    let ⟨n, hn⟩ := exists_nat_ge r; ⟨n, by exact_mod_cast hn⟩
+#align int.comap_coe_at_top Int.comap_cast_atTop
+
+@[simp]
+theorem Int.comap_cast_atBot [StrictOrderedRing R] [Archimedean R] :
+    comap ((↑) : ℤ → R) atBot = atBot :=
+  comap_embedding_atBot (fun _ _ => Int.cast_le) fun r =>
+    let ⟨n, hn⟩ := exists_nat_ge (-r)
+    ⟨-n, by simpa [neg_le] using hn⟩
+#align int.comap_coe_at_bot Int.comap_cast_atBot
+
+theorem tendsto_int_cast_atTop_iff [StrictOrderedRing R] [Archimedean R] {f : α → ℤ}
+    {l : Filter α} : Tendsto (fun n => (f n : R)) l atTop ↔ Tendsto f l atTop := by
+  rw [← @Int.comap_cast_atTop R, tendsto_comap_iff]; rfl
+#align tendsto_coe_int_at_top_iff tendsto_int_cast_atTop_iff
+
+theorem tendsto_int_cast_atBot_iff [StrictOrderedRing R] [Archimedean R] {f : α → ℤ}
+    {l : Filter α} : Tendsto (fun n => (f n : R)) l atBot ↔ Tendsto f l atBot := by
+  rw [← @Int.comap_cast_atBot R, tendsto_comap_iff]; rfl
+#align tendsto_coe_int_at_bot_iff tendsto_int_cast_atBot_iff
+
+theorem tendsto_int_cast_atTop_atTop [StrictOrderedRing R] [Archimedean R] :
+    Tendsto ((↑) : ℤ → R) atTop atTop :=
+  tendsto_int_cast_atTop_iff.2 tendsto_id
+#align tendsto_coe_int_at_top_at_top tendsto_int_cast_atTop_atTop
+
+@[simp]
+theorem Rat.comap_cast_atTop [LinearOrderedField R] [Archimedean R] :
+    comap ((↑) : ℚ → R) atTop = atTop :=
+  comap_embedding_atTop (fun _ _ => Rat.cast_le) fun r =>
+    let ⟨n, hn⟩ := exists_nat_ge r; ⟨n, by simpa⟩
+#align rat.comap_coe_at_top Rat.comap_cast_atTop
+
+@[simp] theorem Rat.comap_cast_atBot [LinearOrderedField R] [Archimedean R] :
+    comap ((↑) : ℚ → R) atBot = atBot :=
+  comap_embedding_atBot (fun _ _ => Rat.cast_le) fun r =>
+    let ⟨n, hn⟩ := exists_nat_ge (-r)
+    ⟨-n, by simpa [neg_le] ⟩
+#align rat.comap_coe_at_bot Rat.comap_cast_atBot
+
+theorem tendsto_rat_cast_atTop_iff [LinearOrderedField R] [Archimedean R] {f : α → ℚ}
+    {l : Filter α} : Tendsto (fun n => (f n : R)) l atTop ↔ Tendsto f l atTop := by
+  rw [← @Rat.comap_cast_atTop R, tendsto_comap_iff]; rfl
+#align tendsto_coe_rat_at_top_iff tendsto_rat_cast_atTop_iff
+
+theorem tendsto_rat_cast_atBot_iff [LinearOrderedField R] [Archimedean R] {f : α → ℚ}
+    {l : Filter α} : Tendsto (fun n => (f n : R)) l atBot ↔ Tendsto f l atBot := by
+  rw [← @Rat.comap_cast_atBot R, tendsto_comap_iff]; rfl
+#align tendsto_coe_rat_at_bot_iff tendsto_rat_cast_atBot_iff
+
+-- porting note: new lemma
+theorem atTop_antitone_basis_of_archimedean [StrictOrderedSemiring R] [Archimedean R] :
```

Comment:
```suggestion
theorem atTop_hasAntitoneBasis_of_archimedean [StrictOrderedSemiring R] [Archimedean R] :
```
Seems like this would be better, no? (Same for countable basis below.)